created shell_sort program

Rishabh062 opened the pull request:
Added shell sort program well written in cpp language.

Rishabh062 committed:
created shell_sort program

Added shell sort program well written in cpp language.

SAY-droid427 commented:
Hey! @Rishabh062  Please use multiline comment on the top of the program to mention the problem and a short algorithm.Include time and space complexity if possible.

Rishabh062 commented:
@SAY-droid427 ok I will do that.

Rishabh062 committed:
Update shell_sort.cpp

Added multiple comments which has made code easier to understand.
Added:
1) About problem
2) Time and Space complexity
3) Some algorithm about shell sort
